tasks/administer-cluster/kubelet-in-userns: update outdated information (#56638)
kubernetes/website: content/en/docs/tasks/administer-cluster/kubelet-in-userns.md

* [Rootless Docker](https://minikube.sigs.k8s.io/docs/drivers/docker/)
* [Rootless Podman](https://minikube.sigs.k8s.io/docs/drivers/podman/)

### Usernetes

{{% thirdparty-content %}}

[Usernetes](https://github.com/rootless-containers/usernetes) also supports running Kubernetes inside Rootless Docker, Rootless Podman, or Rootless nerdctl.

> Unlike `kind` and `minikube`, Usernetes supports composing a cluster using multiple Docker/Podman/nerdctl nodes using VXLAN.

Reviewer: nit: the page uses bare text for tool names like kind and minikube. Backticking them here is inconsistent either remove backticks here, or align all other places

```suggestion
Unlike kind and minikube, Usernetes supports composing a cluster using multiple Docker/Podman/nerdctl nodes using VXLAN.
```

See the [Usernetes documentation](https://github.com/rootless-containers/usernetes/blob/master/README.md) for usage.

> ## Running Kubernetes inside Kubernetes, as UserNS-enabled pods

Reviewer: Missing `{{% thirdparty-content %}}` — all sibling sections have one. Also, "UserNS-enabled" is informal; the rest of the page spells out "user namespace".

```suggestion
## Running Kubernetes inside Kubernetes, as user namespace pods

{{% thirdparty-content single="true" %}}
```

Author: Applied the suggestion, but do we need to repeat `thirdparty-content` 4 times?
Can we just show it only once?

Reviewer: it was already there multiple times. Since the notice applicable to specific sections, it seems fine to do. Inconsistent application of it is worse.

Maybe use of `{{% thirdparty-content single="true" %}}` is better here. Here is how it renders: https://deploy-preview-56638--kubernetes-io-main-staging.netlify.app/docs/tasks/administer-cluster/kubelet-in-userns/

I think ideally there should be a way to say this on top of the page and mention all third-party tools. But I didn't find a way to do it and marking the whole article is unreasoinable

Author: > Maybe use of {{% thirdparty-content single="true" %}} is better here

Done

See [Running K3s with Rootless mode](https://rancher.com/docs/k3s/latest/en/advanced/#running-k3s-with-rootless-mode-experimental) for the usage.

### Usernetes (until 2023)

> [Usernetes](https://github.com/rootless-containers/usernetes) supported running rootless Kubernetes directly on a host until 2023.

Reviewer: ```suggestion
The gen1 branch of [Usernetes](https://github.com/rootless-containers/usernetes) supported running rootless Kubernetes directly on a host. This mode was archived in 2023.
```
